rtrim() expects parameter 1 to be string, bool given

When running phive on Windows 10, using Powershell, I'm getting the error:

`rtrim() expects parameter 1 to be string, bool given`

Here is the full output:

```sh
λ php .\phive.phar install phpunit@8.0
Phive 0.13.2 - Copyright (C) 2015-2019 by Arne Blankerts, Sebastian Heuer and Contributors
Fetching repository list
Downloading https://phar.io/data/repositories.xml
Downloading https://phar.phpunit.de/phive.xml
Downloading https://phar.phpunit.de/phpunit-8.0.6.phar
Downloading https://phar.phpunit.de/phpunit-8.0.6.phar.asc
Downloading key 4AA394086372C20A
Trying to connect to keys.openpgp.org (37.218.245.50)
Downloading https://keys.openpgp.org/pks/lookup?op=get&options=mr&search=0x4AA394086372C20A
Successfully downloaded key.

        Fingerprint: D840 6D0D 8294 7747 2937 7831 4AA3 9408 6372 C20A

        Sebastian Bergmann <sb@sebastian-bergmann.de>
        Sebastian Bergmann <sebastian.bergmann@thephp.cc>
        Sebastian Bergmann <sebastian@php.net>
        Sebastian Bergmann <sebastian@phpunit.de>
        Sebastian Bergmann <sebastian@thephp.cc>

        Created: 2014-07-19

Import this key? [y|N] [ERROR]    An error occurred while processing your request:

          rtrim() expects parameter 1 to be string, bool given

          #0 src/shared/cli/input/ConsoleInput.php(23)
          #1 src/shared/cli/input/ConsoleInput.php(23): rtrim()
          #2 src/services/key/KeyService.php(64): PharIo\Phive\Cli\ConsoleInput->confirm()
          #3 src/services/key/KeyService.php(48): PharIo\Phive\KeyService->allowedToImport()
          #4 src/services/signature/gpg/GnupgSignatureVerifier.php(27): PharIo\Phive\KeyService->importKey()
          #5 src/services/phar/PharDownloader.php(84): PharIo\Phive\GnupgSignatureVerifier->verify()
          #6 src/services/phar/PharDownloader.php(44): PharIo\Phive\PharDownloader->verifySignature()
          #7 src/services/phar/PharService.php(22): PharIo\Phive\PharDownloader->download()
          #8 src/services/phar/InstallService.php(45): PharIo\Phive\PharService->getPharFromRelease()
          #9 src/commands/install/InstallCommand.php(45): PharIo\Phive\InstallService->execute()
          #10 src/commands/install/InstallCommand.php(37): PharIo\Phive\InstallCommand->installRequestedPhar()
          #11 src/shared/cli/Runner.php(193): PharIo\Phive\InstallCommand->execute()
          #12 src/shared/cli/Runner.php(60): PharIo\Phive\Cli\Runner->execute()
          #13 (313): PharIo\Phive\Cli\Runner->run()
          #14 {main}

          Environment: PHP 7.4.0
          Phive Version: 0.13.2

          This should not have happened and is most likely a bug.
          Please report it at https://github.com/phar-io/phive/issues, make sure you include
          the full output of this error message. Thank you!
```

I fired up my Win 10 VM to try this and fail to reproduce it:


## cmd.exe

```
C:\Users\TheSeer\Desktop\xx>php phive-0.13.2.phar install phpunit@8.0
Phive 0.13.2 - Copyright (C) 2015-2019 by Arne Blankerts, Sebastian Heuer and Contributors
Fetching repository list
Downloading https://phar.io/data/repositories.xml
Downloading https://phar.phpunit.de/phive.xml
Downloading https://phar.phpunit.de/phpunit-8.0.6.phar
Downloading https://phar.phpunit.de/phpunit-8.0.6.phar.asc
Downloading key 4AA394086372C20A
Trying to connect to keys.openpgp.org (37.218.245.50)
Downloading https://keys.openpgp.org/pks/lookup?op=get&options=mr&search=0x4AA394086372C20A
Successfully downloaded key.

        Fingerprint: D840 6D0D 8294 7747 2937 7831 4AA3 9408 6372 C20A

        Sebastian Bergmann <sb@sebastian-bergmann.de>
        Sebastian Bergmann <sebastian.bergmann@thephp.cc>
        Sebastian Bergmann <sebastian@php.net>
        Sebastian Bergmann <sebastian@phpunit.de>
        Sebastian Bergmann <sebastian@thephp.cc>

        Created: 2014-07-19

Import this key? [y|N] n
[ERROR]   User declined import of key 4AA394086372C20A
[ERROR]   Signature could not be verified
[ERROR]   [GNUPG:] NEWSIG
[ERROR]   [GNUPG:] ERRSIG 4AA394086372C20A 1 10 00 1553608850 9 -
[ERROR]   [GNUPG:] NO_PUBKEY 4AA394086372C20A
```

## Powershell
```
PS C:\Users\TheSeer\Desktop\xx> php .\phive-0.13.2.phar install phpunit@8.0
Phive 0.13.2 - Copyright (C) 2015-2019 by Arne Blankerts, Sebastian Heuer and Contributors
Downloading https://phar.phpunit.de/phpunit-8.0.6.phar
Downloading https://phar.phpunit.de/phpunit-8.0.6.phar.asc
Downloading key 4AA394086372C20A
Trying to connect to keys.openpgp.org (37.218.245.50)
Downloading https://keys.openpgp.org/pks/lookup?op=get&options=mr&search=0x4AA394086372C20A
Successfully downloaded key.

        Fingerprint: D840 6D0D 8294 7747 2937 7831 4AA3 9408 6372 C20A

        Sebastian Bergmann <sb@sebastian-bergmann.de>
        Sebastian Bergmann <sebastian.bergmann@thephp.cc>
        Sebastian Bergmann <sebastian@php.net>
        Sebastian Bergmann <sebastian@phpunit.de>
        Sebastian Bergmann <sebastian@thephp.cc>

        Created: 2014-07-19

Import this key? [y|N] n
[ERROR]   User declined import of key 4AA394086372C20A
[ERROR]   Signature could not be verified
[ERROR]   [GNUPG:] NEWSIG
[ERROR]   [GNUPG:] ERRSIG 4AA394086372C20A 1 10 00 1553608850 9 -
[ERROR]   [GNUPG:] NO_PUBKEY 4AA394086372C20A

PS C:\Users\TheSeer\Desktop\xx>
```

Works for me? I do realize we should catch the potential boolean return value but I fail to see how this can happen ;)

--- a/src/shared/cli/input/ConsoleInput.php
+++ b/src/shared/cli/input/ConsoleInput.php
@@ -27,7 +27,13 @@ class ConsoleInput implements Input {
 
         do {
             $this->output->writeText(\rtrim($message) . \sprintf(' [%s|%s] ', $yesOption, $noOption));
-            $response = \strtolower(\rtrim(\fgets($this->inputStream)));
+            $input = \fgets($this->inputStream);
+
+            if ($input === false) {
+                throw new RunnerException('Needs tty to be able to confirm');
+            }
+
+            $response = \strtolower(\rtrim($input));
         } while (!\in_array($response, ['y', 'n', ''], true));
 
         if ($response === '') {

--- a/tests/unit/shared/cli/ConsoleInputTest.php
+++ b/tests/unit/shared/cli/ConsoleInputTest.php
@@ -3,6 +3,7 @@ namespace PharIo\Phive;
 
 use PharIo\Phive\Cli\ConsoleInput;
 use PharIo\Phive\Cli\Output;
+use PharIo\Phive\Cli\RunnerException;
 use PHPUnit\Framework\TestCase;
 
 /**
@@ -84,6 +85,27 @@ class ConsoleInputTest extends TestCase {
 
         $input = new ConsoleInput($output, $inputStream);
         $this->assertSame($default, $input->confirm('foo?', $default));
+    }
+
+    public function testOnNonInteractive(): void {
+        $output = $this->getOutputMock();
+        $output->expects($this->once())
+            ->method('writeText')
+            ->with('foo? [Y|n] ');
+
+        /*
+         * Emulate a non-interactive shell by not writing anything
+         *
+         * In a real shell it can be emulate with `echo -n | phive install ...`
+         */
+        $inputStream = \fopen('php://memory', 'r');
+
+        $input = new ConsoleInput($output, $inputStream);
+
+        $this->expectException(RunnerException::class);
+        $this->expectExceptionMessage('Needs tty to be able to confirm');
+
+        $input->confirm('foo?');
     }
 
     /**

